models/models.js

```diff
@@ -41,3 +41,37 @@ exports.selectUsers = () => {
     return rows;
   });
 };
+
+exports.selectArticles = () => {
+  return db
+    .query(
+      "SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.created_at, articles.votes, (SELECT COUNT(*) :: INT FROM comments WHERE comments.article_id = articles.article_id) AS comment_count FROM articles;"
+    )
+    .then(({ rows }) => {
+      return rows;
+    });
+};
+
+exports.selectCommentsByArticleId = (article_id) => {
+  return db
+    .query("SELECT * FROM comments WHERE comments.article_id = $1;", [
+      article_id,
+    ])
+    .then(({ rows }) => {
+      console.log(rows);
+      return rows;
+    });
+};
+
+exports.selectArticleByIdForComments = async (article_id) => {
+  const {
+    rows: [article],
+  } = await db.query("SELECT * FROM articles WHERE article_id = $1;", [
+    article_id,
+  ]);
+
+  if (!article) {
+    return Promise.reject({ status: 404, msg: "Invalid article id" });
+  }
+  return article;
```

Comment:
If this function's sole purpose is to check if the article actually exists, it doesn't need to return anything.

Reply:
Ahh i see, i will remove the return. thanks